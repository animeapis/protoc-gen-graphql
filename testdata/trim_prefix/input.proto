syntax = "proto3";

package protoc_gen_graphql.test.trim_prefix;

import "graphql/options.proto";

option go_package = "github.com/animeapis/protoc-gen-graphql/testdata;testdata";

service Service {
  rpc Method(Message) returns (Message) {
    option (graphql.method) = { operation: "query" };
  }
}

message Message {
  string name = 1;

  oneof oneof_field {
    string first = 2;
    string second = 3;
  }
}

enum Enum {
  ZERO = 0;
}
