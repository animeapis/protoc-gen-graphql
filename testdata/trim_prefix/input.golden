# DO NOT EDIT! Generated by protoc-gen-graphql.

type TestTrimPrefix_Service_Query {
  method(input: TestTrimPrefix_MessageInput!): TestTrimPrefix_Message!
}

type TestTrimPrefix_Message {
  name: String!
  oneofField: TestTrimPrefix_Message_OneofFieldOneof
}

"""
`TestTrimPrefix_Message_OneofFieldOneof` represents the `oneof_field` oneof in `protoc_gen_graphql.test.trim_prefix.Message`.
"""
union TestTrimPrefix_Message_OneofFieldOneof = TestTrimPrefix_Message_OneofFieldOneof_First | TestTrimPrefix_Message_OneofFieldOneof_Second

"""
`TestTrimPrefix_Message_OneofFieldOneof_First` represents the `first` oneof field in `protoc_gen_graphql.test.trim_prefix.Message`.
"""
type TestTrimPrefix_Message_OneofFieldOneof_First {
  _typename: String
  first: String!
}

"""
`TestTrimPrefix_Message_OneofFieldOneof_Second` represents the `second` oneof field in `protoc_gen_graphql.test.trim_prefix.Message`.
"""
type TestTrimPrefix_Message_OneofFieldOneof_Second {
  _typename: String
  second: String!
}

input TestTrimPrefix_MessageInput {
  name: String
  oneofField: TestTrimPrefix_Message_OneofFieldOneofInput
}

input TestTrimPrefix_Message_OneofFieldOneofInput {
  first: String
  second: String
}

enum TestTrimPrefix_Enum {
  ZERO
}
